playbooks/sample-monitoring-sap.yml

```diff
@@ -0,0 +1,26 @@
+---
+- name:  sap monitoring configuration
+  hosts: 127.0.0.1
+  connection: local
+  gather_facts: no
+  tasks:
+
+    - name: Execute the  role monitoring_sap
+      # include_role:
+      import_role:
+        name: { role: ../roles/monitoring_sap }
+      when: sap_monitoring_action == "add"
+
+      ##  Delete a SAP System from monitoring
+    - name: Check existence of the variables defined in roles/monitoring_sap/vars/main.yml
```

Comment:
this check should be handeled internally which is required for deleting a system,

Reply:
feasible when I had several roles.
With one role only excuting several role-functions I split into:
- main.yml: only with action "add"  (callling 5 task-files)
- NOT calling main.yml: action "delete"

moving this to main.yml means:
main.yml becomes functional "samle-monitoring-sap.yml" calling  main2.yml 

You will see as soon as you move the 6 SAP-reparing-roles into one role.
I moved 5 exiting monitoring roles into one role.

Reply:
With using block it may happen.
The bad desing would remain: Calling all taks and skipping the  tasks for the unused action-paramater.

Reply:
That's how it has to. Skipping is okay and is considered good

Reply:
now 1 main.yml with blocks
minimal sample-monitoring-sap.yml  with calling the role only.